service/history/shardContext.go

```diff
@@ -127,6 +129,7 @@ type (
 
 		// exist only in memory
 		standbyClusterCurrentTime map[string]time.Time
+		shardOwnershipChanged     bool
```

Comment:
ownershipChanged is a bit confusing to me - (1) does it mean this shard is no longer valid because ownership has moved to another host or (2) does it mean this shard is loaded because of a previous ownership change. If I understand right, you want (2) ? Is that right ? If that's the case, how do you feel about storing "previousOwner" here instead of this flag.

Reply:
This flag means that while we were trying to acquire a shard, record that we've loaded from the database had a different owner than the current host's identity. Hence it means that shard ownership has changed and this is the first time that we've discovered it.